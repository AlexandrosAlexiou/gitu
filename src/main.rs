use std::{
    error::Error,
    io::{stderr, BufWriter},
};

use clap::Parser;
use crossterm::{
    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
    ExecutableCommand,
};
use ratatui::{prelude::CrosstermBackend, Terminal};

pub fn main() -> Result<(), Box<dyn Error>> {
    let mut terminal = Terminal::new(CrosstermBackend::new(BufWriter::new(stderr())))?;
    terminal.hide_cursor()?;
    enable_raw_mode()?;
    stderr().execute(EnterAlternateScreen)?;

    let result = gitu::run(gitu::cli::Args::parse(), &mut terminal);

    stderr().execute(LeaveAlternateScreen)?;
    disable_raw_mode()?;

    result?;

    Ok(())
}
