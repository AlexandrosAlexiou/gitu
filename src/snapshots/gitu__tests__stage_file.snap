---
source: src/main.rs
expression: terminal.backend().buffer()
---
Buffer {
    area: Rect { x: 0, y: 0, width: 80, height: 30 },
    content: [
        " On branch master                                                               ",
        "                                                                                ",
        " No commits yet                                                                 ",
        "                                                                                ",
        " Changes to be committed:                                                       ",
        "   (use "git rm --cached <file>..." to unstage)                                 ",
        " new file:   new-file                                                           ",
        "                                                                                ",
        "                                                                                ",
        " Staged changes (1)                                                             ",
        "ðŸ¢’new-file                                                                       ",
        "                                                                                ",
        " Recent commits                                                                 ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
    ],
    styles: [
        x: 0, y: 0, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 6, fg: Green, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 6, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 9, fg: Rgb(216, 166, 87), bg: Reset, underline: Reset, modifier: BOLD,
        x: 19, y: 9, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 10, fg: Reset, bg: Rgb(80, 73, 69), underline: Reset, modifier: NONE,
        x: 1, y: 10, fg: Rgb(211, 134, 155), bg: Rgb(80, 73, 69), underline: Reset, modifier: BOLD,
        x: 9, y: 10, fg: Reset, bg: Rgb(80, 73, 69), underline: Reset, modifier: NONE,
        x: 0, y: 11, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 12, fg: Rgb(216, 166, 87), bg: Reset, underline: Reset, modifier: BOLD,
        x: 15, y: 12, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
